### PR TITLE
Fixing responsiveness for order filter

### PR DESCRIPTION
**Pull request by samkevin1**
Resolve https://github.com/opencollective/opencollective/issues/7226

# Description

- Made the order filter responsive on the new dashboard.
- Removed the order filter from always being at the end of the section, so it breaks correctly to the next line when needed.

# Screenshots

image

image

### CODE REVIEW

**Review by Betree — COMMENTED** [review]
I'll let @gustavlrsn review that one

File: components/dashboard/filters/Filterbar.tsx
Comment (gustavlrsn) [inline components/dashboard/filters/Filterbar.tsx:120]: While we might want to improve how the filters wrap and break together with the order filter, I still think we want to keep the order filter aligned to the right, to give it a consistent expected position across all pages and more importantly have the concept of **sorting** slightly set apart from the concept of **filtering**.

One thing to fix here though, to allow truncating long Order filter labels on small screens we can set "w-full" here:

```suggestion
          <div className="flex w-full flex-1 justify-end">
```

File: components/dashboard/filters/OrderFilter.tsx
Comment (gustavlrsn) [inline components/dashboard/filters/OrderFilter.tsx:52]: Nice fix to use "fit-content" here instead of a fixed value 👍 . 

However, we should set it as the max-width, since if the language string is really long and you are on a small screen we'd want the layout to limit the width instead.

```suggestion
      <Button size="sm" variant="outline" asChild className="max-w-fit rounded-full">
```

File: components/dashboard/filters/OrderFilter.tsx
Comment (gustavlrsn) [inline components/dashboard/filters/OrderFilter.tsx:55]: Found an issue with the text truncation here, solved by this setting 'overflow-hidden' on the parent of the text to be truncated, also making sure to not shrink the icon.

```suggestion
          <div className="flex items-center gap-2 overflow-hidden font-medium">
            <Icon className="-ml-0.5 h-5 w-5 shrink-0 text-slate-400" aria-hidden="true" />
```

**Review by gustavlrsn — CHANGES_REQUESTED** [review]
@samkevin1 thank you for the contribution! Added some comments/suggestions on things to address before merging the PR

File: components/dashboard/filters/Filterbar.tsx
Comment (samkevin1) [reply components/dashboard/filters/Filterbar.tsx:120]: I just tested this change (`w-full`) and it makes the sorting filter always break to the next line. Is that expected?

image

Comment (samkevin1) [reply components/dashboard/filters/Filterbar.tsx]: Actually, nevermind. The css was wrong when testing this.

**Comment by samkevin1** [comment]
@gustavlrsn Made the changes. Can you take a look again?

**Review by gustavlrsn — APPROVED** [review]
@samkevin1 Thank you, looks good to me!

**Pull request merged**
